zomes/hc_zome_trust_atom/src/lib.rs

```diff
@@ -16,15 +16,36 @@ use std::collections::BTreeMap;
 // public for sweettest; TODO can we fix this:
 pub mod trust_atom;
 pub use crate::trust_atom::*;
-pub mod test_helpers;
+pub mod trust_graph;
+pub use crate::trust_graph::*;
+pub(crate) mod agents;
+pub(crate) mod utils;
+pub(crate) mod test_helpers;
 
 entry_defs![test_helpers::StringTarget::entry_def(), Extra::entry_def()];
 
 // INPUT TYPES
 
+#[derive(Serialize, Deserialize, SerializedBytes, Debug, Clone)]
+pub enum HashType {
+  Agent(AgentPubKey),
+  Entry(AgentPubKey),
+  //Header(HeaderHash)
+    //Entity( //raw bytes hash )
+}
+
+#[derive(Serialize, Deserialize, SerializedBytes, Debug, Clone)]
+pub enum TrustGraphInput {
+  Mine(LinkTag), // my TrustGraph (list of all my Trust Atoms optionally filtered by tag)
+  MyRollup(LinkTag), // Mine + Rollup (other agents TG whom Ive given TAs)
+  //Agent(AgentPubKey, LinkTag) 
+  //AgentRollup(AgentPubKey, LinkTag) // another agent's TrustGraph + their Rollup
+}
+
 #[derive(Serialize, Deserialize, SerializedBytes, Debug, Clone)]
 pub struct TrustAtomInput {
   pub target: EntryHash, // TODO maybe target_entry_hash?
+  pub label: Option<String>,
```

Comment:
maybe better word than label -- more like type

Reply:
we settled on prefix